From 6bccc4a1f862135939c2d420a5a3f3a20031b3c8 Mon Sep 17 00:00:00 2001
From: Sam Nelson <sam.nelson@ti.com>
Date: Fri, 11 Nov 2016 10:32:48 -0500
Subject: [PATCH 1/4] TEMP: ARM: dts: keystone-k2hk: Split sram nodes to allow
 section for CMEM use

Currently the reg entry in kernel dtb reserves the whole MSMC ram range
to be used only through the mmio-sram driver.

Given that other modules (like CMEM) also currently reserve memory from
MSMC, the reg entry which takes over the whole memory breaks those
functionality.
This patch will allow module like CMEM to reserve and use a non-
overlapping section of memory not marked reserved by the sub nodes here.
And is a workaround to unblock functionality.
Final solution is to implement functionality in CMEM driver to use sram
sub nodes directly.

Resolves current issue with CMEM module reporting error allocating MSMC
memory.

Signed-off-by: Sam Nelson <sam.nelson@ti.com>
---
 arch/arm/boot/dts/keystone-k2hk.dtsi | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/keystone-k2hk.dtsi b/arch/arm/boot/dts/keystone-k2hk.dtsi
index edca6aa..5136160 100644
--- a/arch/arm/boot/dts/keystone-k2hk.dtsi
+++ b/arch/arm/boot/dts/keystone-k2hk.dtsi
@@ -111,8 +111,8 @@
 
 		msm_ram: msmram@0c000000 {
 			compatible = "mmio-sram";
-			reg = <0x0c000000 0x600000>;
-			ranges = <0x0 0x0c000000 0x600000>;
+			reg = <0x0c000000 0x100000>;
+			ranges = <0x0 0x0c000000 0x100000>;
 			#address-cells = <1>;
 			#size-cells = <1>;
 
@@ -120,9 +120,17 @@
 				compatible = "ti,keystone-dsp-msm-ram";
 				reg = <0x0 0x80000>;
 			};
+		};
+
+		msm_ram1: msmram1@0c580000 {
+			compatible = "mmio-sram";
+			reg = <0x0c580000 0x80000>;
+			ranges = <0x0 0x0c580000 0x80000>;
+			#address-cells = <1>;
+			#size-cells = <1>;
 
-			sram-bm@5f0000 {
-				reg = <0x5f0000 0x8000>;
+			sram-bm@70000 {
+				reg = <0x70000 0x8000>;
 			};
 		};
 
-- 
1.9.1

